fw/application/src/app/settings/scene/settings_scene_language.c

```diff
@@ -7,59 +7,29 @@
 #include "utils2.h"
 #include "version2.h"
 
-enum settings_sleep_timeout_menu_t {
-    SETTINGS_LANGUAGE_ZH_HANS,
-    SETTINGS_LANGUAGE_EN_US,
-    SETTINGS_LANGUAGE_ZH_TW,
-    SETTINGS_LANGUAGE_ES_ES,
-    SETTINGS_LANGUAGE_HU_HU,
-    SETTINGS_LANGUAGE_KO_KR,
-    SETTINGS_LANGUAGE_MENU_EXIT,
-};
+int SetLangOpt;
 
 static void settings_scene_language_list_view_on_selected(mui_list_view_event_t event, mui_list_view_t *p_list_view,
                                                           mui_list_item_t *p_item) {
+
     app_settings_t *app = p_list_view->user_data;
     uint32_t selection = (uint32_t)p_item->user_data;
     settings_data_t *p_settings = settings_get_data();
-    switch (selection) {
-    case SETTINGS_LANGUAGE_ZH_HANS:
-        p_settings->language = LANGUAGE_ZH_HANS;
-        setLanguage(p_settings->language);
-        break;
-
-    case SETTINGS_LANGUAGE_EN_US:
-        p_settings->language = LANGUAGE_EN_US;
-        setLanguage(p_settings->language);
-        break;
-
-    case SETTINGS_LANGUAGE_ZH_TW:
-        p_settings->language = LANGUAGE_ZH_TW;
-        setLanguage(p_settings->language);
-        break;
-
-    case SETTINGS_LANGUAGE_ES_ES:
-        p_settings->language = LANGUAGE_ES_ES;
-        setLanguage(p_settings->language);
-        break;
-
-    case SETTINGS_LANGUAGE_HU_HU:
-        p_settings->language = LANGUAGE_HU_HU;
+    if (selection < LANGUAGE_COUNT){
+        p_settings->language = selection;
         setLanguage(p_settings->language);
-        break;
     }
+
     mui_scene_dispatcher_previous_scene(app->p_scene_dispatcher);
 }
 
 void settings_scene_language_on_enter(void *user_data) {
 
     app_settings_t *app = user_data;
-    mui_list_view_add_item(app->p_list_view, 0xe105, "简体中文", (void *)SETTINGS_LANGUAGE_ZH_HANS);
-    mui_list_view_add_item(app->p_list_view, 0xe105, "繁體中文(臺灣)", (void *)SETTINGS_LANGUAGE_ZH_TW);
-    mui_list_view_add_item(app->p_list_view, 0xe105, "English", (void *)SETTINGS_LANGUAGE_EN_US);
-    mui_list_view_add_item(app->p_list_view, 0xe105, getLangDesc(LANGUAGE_ES_ES), (void *)SETTINGS_LANGUAGE_ES_ES);
-    mui_list_view_add_item(app->p_list_view, 0xe105, getLangDesc(LANGUAGE_HU_HU), (void *)SETTINGS_LANGUAGE_HU_HU);
-    mui_list_view_add_item(app->p_list_view, 0xe069, getLangString(_L_BACK), (void *)SETTINGS_LANGUAGE_MENU_EXIT);
+    for (SetLangOpt = 0; SetLangOpt < LANGUAGE_COUNT; SetLangOpt++){
```

Comment:
`SetlangOpt` can be simple to `i` as c program style. 
And dont forget to declare the variable type, e.g. `for(uint8_t i = 0; i<LANGUAGE_COUNT; i++){`

Reply:
Yeah at first I did used `i` but want to be more specific `i` is ok.

I put the definition outside of the `for` because the last item `BACK` need it to be the last one and I used the last value of `i` but also can we use the `LANGUAGE_COUNT` definition. what do you think?

Reply:
The user data of last `BACK` is not used, you can set macro `NULL_USER_DATA` instead.

Reply:
excellent working on that!